client/reader/site-stream/featured.jsx

```diff
@@ -24,8 +26,37 @@ export default React.createClass( {
 
 	getStateFromStores( store ) {
 		store = store || this.props.store;
+
+		let postKeys = store.get();
+		let posts = postKeys.map( postKey => {
+			let post = FeedPostStore.get( postKey ),
+				originalPost,
+				isDiscoverPost;
+
+			if ( ! post || post._state === 'minimal' ) {
+				FeedPostStoreActions.fetchPost( postKey );
+			} else {
+				isDiscoverPost = post && DiscoverHelper.isDiscoverPost( post );
+
+				let isDiscoverSitePick = post && isDiscoverPost && DiscoverHelper.isDiscoverSitePick( post );
+
+				if ( isDiscoverPost && ! isDiscoverSitePick && get( post, 'discover_metadata.featured_post_wpcom_data.blog_id' ) ) {
+					originalPost = FeedPostStore.get( {
+						blogId: post.discover_metadata.featured_post_wpcom_data.blog_id,
+						postId: post.discover_metadata.featured_post_wpcom_data.post_id
+					} );
+				}
+			}
```

Comment:
I agree; the method was pretty clunky and unwieldy. 87c46b4 to simplify things a lot.